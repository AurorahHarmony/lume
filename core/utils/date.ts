import { Temporal } from "../../deps/temporal.ts";

/**
 * Returns the date of the git commit that created or modified the file.
 * Thanks to https://github.com/11ty/eleventy/blob/8dd2a1012de92c5ee1eab7c37e6bf1b36183927e/src/Util/DateGitLastUpdated.js
 */
export function getGitDate(
  type: "created" | "modified",
  file: string,
): Date | undefined {
  const args = type === "created"
    ? ["log", "--diff-filter=A", "--follow", "-1", "--format=%at", "--", file]
    : ["log", "-1", "--format=%at", "--", file];

  const { stdout, success } = new Deno.Command("git", { args }).outputSync();

  if (!success) {
    return;
  }
  const str = new TextDecoder().decode(stdout);
  const timestamp = parseInt(str) * 1000;

  if (timestamp) {
    return new Date(timestamp);
  }
}

/** Parse a string to Date */
export function parseDate(date: string): Date {
  return new Date(getZonedDateTime(date).epochMilliseconds);
}

/** Parse a string to a zoned datetime */
export function getZonedDateTime(
  date: string,
  timezone = "UTC",
): Temporal.ZonedDateTime {
  try {
    return Temporal.Instant.from(date).toZonedDateTimeISO(timezone);
  } catch {
    return Temporal.PlainDateTime.from(date).toZonedDateTime(timezone);
  }
}
