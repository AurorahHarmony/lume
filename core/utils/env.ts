export function env<T>(name: string): T | undefined {
  const value = Deno.env.get(name);

  if (typeof value === "undefined") {
    return undefined;
  }

  switch (value.toLowerCase()) {
    case "true":
    case "on":
    case "1":
      return true as T;

    case "false":
    case "off":
    case "0":
      return false as T;

    default:
      return value as T;
  }
}
