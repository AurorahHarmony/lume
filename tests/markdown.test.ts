import { assertSiteSnapshot, build, getSite } from "./utils.ts";

Deno.test("Build a markdown site", async (t) => {
  const site = getSite({
    src: "markdown",
    location: new URL("https://example.com/blog"),
  });

  await build(site);
  await assertSiteSnapshot(t, site);
});
