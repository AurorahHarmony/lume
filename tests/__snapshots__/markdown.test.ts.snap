export const snapshot = {};

snapshot[`Build a markdown site 1`] = `
{
  formats: [
    {
      engines: 1,
      ext: ".tmpl.ts",
      pageLoader: [AsyncFunction: module],
    },
    {
      engines: 1,
      ext: ".tmpl.js",
      pageLoader: [AsyncFunction: module],
    },
    {
      engines: undefined,
      ext: ".tmpl.jsonc",
      pageLoader: [AsyncFunction: json],
    },
    {
      engines: undefined,
      ext: ".tmpl.json",
      pageLoader: [AsyncFunction: json],
    },
    {
      dataLoader: [AsyncFunction: json],
      engines: undefined,
      ext: ".json",
    },
    {
      dataLoader: [AsyncFunction: json],
      engines: undefined,
      ext: ".jsonc",
    },
    {
      engines: 1,
      ext: ".md",
      pageLoader: [AsyncFunction: text],
    },
    {
      engines: 1,
      ext: ".markdown",
      pageLoader: [AsyncFunction: text],
    },
    {
      componentLoader: [AsyncFunction: module],
      dataLoader: [AsyncFunction: module],
      engines: 1,
      ext: ".js",
    },
    {
      componentLoader: [AsyncFunction: module],
      dataLoader: [AsyncFunction: module],
      engines: 1,
      ext: ".ts",
    },
    {
      componentLoader: [AsyncFunction: text],
      engines: 1,
      ext: ".njk",
      includesPath: "_includes",
      pageLoader: [AsyncFunction: text],
    },
    {
      dataLoader: [AsyncFunction: toml],
      engines: undefined,
      ext: ".toml",
      pageLoader: [AsyncFunction: toml],
    },
    {
      dataLoader: [AsyncFunction: yaml],
      engines: undefined,
      ext: ".yaml",
      pageLoader: [AsyncFunction: yaml],
    },
    {
      dataLoader: [AsyncFunction: yaml],
      engines: undefined,
      ext: ".yml",
      pageLoader: [AsyncFunction: yaml],
    },
  ],
}
`;

snapshot[`Build a markdown site 2`] = `[]`;

snapshot[`Build a markdown site 3`] = `
[
  {
    content: "<!DOCTYPE html>
" +
      "<h1>Normal page</h1>
" +
      "<ul>
" +
      "<li>List</li>
" +
      "<li>Of</li>
" +
      "<li>Elements</li>
" +
      "</ul>
",
    data: {
      children: "<h1>Normal page</h1>
" +
        "<ul>
" +
        "<li>List</li>
" +
        "<li>Of</li>
" +
        "<li>Elements</li>
" +
        "</ul>
",
      content: "# Normal page

- List
- Of
- Elements
",
      date: [],
      mergedKeys: [
        "tags",
      ],
      page: [
        "src",
        "data",
      ],
      paginate: "paginate",
      search: [],
      tags: "Array(0)",
      title: "Basic page",
      url: "/basic/",
    },
    src: {
      asset: undefined,
      ext: ".md",
      path: "/basic",
      remote: undefined,
      slug: "basic",
    },
  },
  {
    content: "<!DOCTYPE html>
" +
      '<p>Here is a footnote reference,<sup class="footnote-ref"><a href="#fn1" id="fnref1">[1]</a></sup> and another.<sup class="footnote-ref"><a href="#fn2" id="fnref2">[2]</a></sup></p>
' +
      "<p>belong to the previous footnote.</p>
" +
      '<h4 class="mt-3">Footnotes</h4>
' +
      '<section class="footnotes">
' +
      '<ol class="footnotes-list">
' +
      '<li id="fn1" class="footnote-item"><p>Here is the footnote. <a href="#fnref1" class="footnote-backref">↩︎</a></p>
' +
      "</li>
" +
      \`<li id="fn2" class="footnote-item"><p>Here's one with multiple blocks.</p>
\` +
      '<p>Subsequent paragraphs are indented to show that they <a href="#fnref2" class="footnote-backref">↩︎</a></p>
' +
      "</li>
" +
      "</ol>
" +
      "</section>
",
    data: {
      children: '<p>Here is a footnote reference,<sup class="footnote-ref"><a href="#fn1" id="fnref1">[1]</a></sup> and another.<sup class="footnote-ref"><a href="#fn2" id="fnref2">[2]</a></sup></p>
' +
        "<p>belong to the previous footnote.</p>
" +
        '<h4 class="mt-3">Footnotes</h4>
' +
        '<section class="footnotes">
' +
        '<ol class="footnotes-list">
' +
        '<li id="fn1" class="footnote-item"><p>Here is the footnote. <a href="#fnref1" class="footnote-backref">↩︎</a></p>
' +
        "</li>
" +
        \`<li id="fn2" class="footnote-item"><p>Here's one with multiple blocks.</p>
\` +
        '<p>Subsequent paragraphs are indented to show that they <a href="#fnref2" class="footnote-backref">↩︎</a></p>
' +
        "</li>
" +
        "</ol>
" +
        "</section>
",
      content: "Here is a footnote reference,[^1] and another.[^longnote]
" +
        "
" +
        "[^1]: Here is the footnote.
" +
        "
" +
        "[^longnote]: Here's one with multiple blocks.
" +
        "
" +
        "    Subsequent paragraphs are indented to show that they
" +
        "
" +
        "belong to the previous footnote.
",
      date: [],
      mergedKeys: [
        "tags",
      ],
      page: [
        "src",
        "data",
      ],
      paginate: "paginate",
      search: [],
      tags: "Array(0)",
      url: "/footnote/",
    },
    src: {
      asset: undefined,
      ext: ".md",
      path: "/footnote",
      remote: undefined,
      slug: "footnote",
    },
  },
  {
    content: '<!DOCTYPE html>
<p><a href="#foo" target="_blank">link</a></p>
',
    data: {
      children: '<p><a href="#foo" target="_blank">link</a></p>
',
      content: "[link](#foo){target=_blank}
",
      date: [],
      mergedKeys: [
        "tags",
      ],
      page: [
        "src",
        "data",
      ],
      paginate: "paginate",
      search: [],
      tags: "Array(0)",
      url: "/with-attributes/",
    },
    src: {
      asset: undefined,
      ext: ".md",
      path: "/with-attributes",
      remote: undefined,
      slug: "with-attributes",
    },
  },
  {
    content: "<!DOCTYPE html>
" +
      '<pre><code class="language-html">&lt;p&gt;Example&lt;/p&gt;
' +
      "</code></pre>
" +
      "<pre><code>&lt;p&gt;Example without color highlight&lt;/p&gt;
" +
      "</code></pre>
" +
      "<pre><code>Indented code
" +
      "</code></pre>
",
    data: {
      children: '<pre><code class="language-html">&lt;p&gt;Example&lt;/p&gt;
' +
        "</code></pre>
" +
        "<pre><code>&lt;p&gt;Example without color highlight&lt;/p&gt;
" +
        "</code></pre>
" +
        "<pre><code>Indented code
" +
        "</code></pre>
",
      content: "\`\`\`html
" +
        "<p>Example</p>
" +
        "\`\`\`
" +
        "
" +
        "\`\`\`
" +
        "<p>Example without color highlight</p>
" +
        "\`\`\`
" +
        "
" +
        "    Indented code
",
      date: [],
      mergedKeys: [
        "tags",
      ],
      page: [
        "src",
        "data",
      ],
      paginate: "paginate",
      search: [],
      tags: "Array(0)",
      url: "/with-code/",
    },
    src: {
      asset: undefined,
      ext: ".md",
      path: "/with-code",
      remote: undefined,
      slug: "with-code",
    },
  },
  {
    content: "<!DOCTYPE html>
" +
      "<!-- deno-fmt-ignore-file -->
" +
      "<dl>
" +
      "<dt>This is a definition title</dt>
" +
      "<dd>And this the description</dd>
" +
      "<dd>Other description</dd>
" +
      "<dt>Other title</dt>
" +
      "<dd>And other description</dd>
" +
      "</dl>
",
    data: {
      children: "<!-- deno-fmt-ignore-file -->
" +
        "<dl>
" +
        "<dt>This is a definition title</dt>
" +
        "<dd>And this the description</dd>
" +
        "<dd>Other description</dd>
" +
        "<dt>Other title</dt>
" +
        "<dd>And other description</dd>
" +
        "</dl>
",
      content: "<!-- deno-fmt-ignore-file -->
" +
        "This is a definition title
" +
        ": And this the description
" +
        ": Other description
" +
        "
" +
        "Other title
" +
        ": And other description
",
      date: [],
      mergedKeys: [
        "tags",
      ],
      page: [
        "src",
        "data",
      ],
      paginate: "paginate",
      search: [],
      tags: "Array(0)",
      url: "/with-deflist/",
    },
    src: {
      asset: undefined,
      ext: ".md",
      path: "/with-deflist",
      remote: undefined,
      slug: "with-deflist",
    },
  },
  {
    content: "<!DOCTYPE html>
" +
      "<h1>Module <strong>example</strong></h1>
" +
      "
" +
      '<div><p>Welcome to this <a href="/">page</a></p></div>
',
    data: {
      children: "<h1>Module <strong>example</strong></h1>
" +
        "
" +
        '<div><p>Welcome to this <a href="/">page</a></p></div>
',
      content: "<h1>{{ title | md(true) | safe }}</h1>
" +
        "
" +
        "<div>{{ description | md | safe }}</div>
",
      date: [],
      description: "Welcome to this [page](/)
",
      mergedKeys: [
        "tags",
      ],
      page: [
        "src",
        "data",
      ],
      paginate: "paginate",
      search: [],
      tags: "Array(0)",
      title: "Module **example**",
      url: "/with-filter/",
    },
    src: {
      asset: undefined,
      ext: ".njk",
      path: "/with-filter",
      remote: undefined,
      slug: "with-filter",
    },
  },
  {
    content: "<!DOCTYPE html>
" +
      "<h1>Module example</h1>
" +
      '<p><a href="/">Back to home</a></p>
',
    data: {
      children: '<h1>Module example</h1>
<p><a href="/">Back to home</a></p>
',
      content: "default",
      date: [],
      mergedKeys: [
        "tags",
      ],
      page: [
        "src",
        "data",
      ],
      paginate: "paginate",
      search: [],
      tags: "Array(0)",
      templateEngine: "tmpl.js,md",
      title: "Module example",
      url: "/with-module/",
    },
    src: {
      asset: undefined,
      ext: ".tmpl.js",
      path: "/with-module",
      remote: undefined,
      slug: "with-module",
    },
  },
  {
    content: "<!DOCTYPE html>
" +
      "<!-- deno-fmt-ignore-file -->
" +
      "<h1></h1>
" +
      "<p>Foo</p>
" +
      "<ul>
" +
      '<li>one: <a href="/items/un.html">un</a></li>
' +
      '<li>two: <a href="/items/dous.html">dous</a></li>
' +
      '<li>three: <a href="/items/tres.html">tres</a></li>
' +
      "</ul>
",
    data: {
      children: "<!-- deno-fmt-ignore-file -->
" +
        "<h1></h1>
" +
        "<p>Foo</p>
" +
        "<ul>
" +
        '<li>one: <a href="/items/un.html">un</a></li>
' +
        '<li>two: <a href="/items/dous.html">dous</a></li>
' +
        '<li>three: <a href="/items/tres.html">tres</a></li>
' +
        "</ul>
",
      content: "<!-- deno-fmt-ignore-file -->
" +
        "# {{ title }}
" +
        "
" +
        "Foo
" +
        "
" +
        "{% for title, no in myData %}
" +
        "- {{ title }}: [{{ no }}](/items/{{ no }}.html)
" +
        "{%- endfor %}
",
      date: [],
      mergedKeys: [
        "tags",
      ],
      myData: [
        "one",
        "two",
        "three",
      ],
      page: [
        "src",
        "data",
      ],
      paginate: "paginate",
      search: [],
      tags: "Array(0)",
      templateEngine: "njk,md",
      url: "/with-nunjucks/",
    },
    src: {
      asset: undefined,
      ext: ".md",
      path: "/with-nunjucks",
      remote: undefined,
      slug: "with-nunjucks",
    },
  },
]
`;

snapshot[`Build a markdown with hooks 1`] = `
{
  formats: [
    {
      engines: 1,
      ext: ".tmpl.ts",
      pageLoader: [AsyncFunction: module],
    },
    {
      engines: 1,
      ext: ".tmpl.js",
      pageLoader: [AsyncFunction: module],
    },
    {
      engines: undefined,
      ext: ".tmpl.jsonc",
      pageLoader: [AsyncFunction: json],
    },
    {
      engines: undefined,
      ext: ".tmpl.json",
      pageLoader: [AsyncFunction: json],
    },
    {
      dataLoader: [AsyncFunction: json],
      engines: undefined,
      ext: ".json",
    },
    {
      dataLoader: [AsyncFunction: json],
      engines: undefined,
      ext: ".jsonc",
    },
    {
      engines: 1,
      ext: ".md",
      pageLoader: [AsyncFunction: text],
    },
    {
      engines: 1,
      ext: ".markdown",
      pageLoader: [AsyncFunction: text],
    },
    {
      componentLoader: [AsyncFunction: module],
      dataLoader: [AsyncFunction: module],
      engines: 1,
      ext: ".js",
    },
    {
      componentLoader: [AsyncFunction: module],
      dataLoader: [AsyncFunction: module],
      engines: 1,
      ext: ".ts",
    },
    {
      componentLoader: [AsyncFunction: text],
      engines: 1,
      ext: ".njk",
      includesPath: "_includes",
      pageLoader: [AsyncFunction: text],
    },
    {
      dataLoader: [AsyncFunction: toml],
      engines: undefined,
      ext: ".toml",
      pageLoader: [AsyncFunction: toml],
    },
    {
      dataLoader: [AsyncFunction: yaml],
      engines: undefined,
      ext: ".yaml",
      pageLoader: [AsyncFunction: yaml],
    },
    {
      dataLoader: [AsyncFunction: yaml],
      engines: undefined,
      ext: ".yml",
      pageLoader: [AsyncFunction: yaml],
    },
  ],
}
`;

snapshot[`Build a markdown with hooks 2`] = `[]`;

snapshot[`Build a markdown with hooks 3`] = `
[
  {
    content: "<!DOCTYPE html>
" +
      "<h1>Normal page</h1>
" +
      "<ul>
" +
      "<li>List</li>
" +
      "<li>Of</li>
" +
      "<li>Elements</li>
" +
      "</ul>
",
    data: {
      children: "<h1>Normal page</h1>
" +
        "<ul>
" +
        "<li>List</li>
" +
        "<li>Of</li>
" +
        "<li>Elements</li>
" +
        "</ul>
",
      content: "# Normal page

- List
- Of
- Elements
",
      date: [],
      mergedKeys: [
        "tags",
      ],
      page: [
        "src",
        "data",
      ],
      paginate: "paginate",
      search: [],
      tags: "Array(0)",
      title: "Basic page",
      url: "/basic/",
    },
    src: {
      asset: undefined,
      ext: ".md",
      path: "/basic",
      remote: undefined,
      slug: "basic",
    },
  },
  {
    content: "<!DOCTYPE html>
" +
      '<p>Here is a footnote reference,<sup class="footnote-ref"><a href="#fn1" id="fnref1">[1]</a></sup> and another.<sup class="footnote-ref"><a href="#fn2" id="fnref2">[2]</a></sup></p>
' +
      "<p>belong to the previous footnote.</p>
" +
      '<h4 class="mt-3">Footnotes</h4>
' +
      '<section class="footnotes">
' +
      '<ol class="footnotes-list">
' +
      '<li id="fn1" class="footnote-item"><p>Here is the footnote. <a href="#fnref1" class="footnote-backref">↩︎</a></p>
' +
      "</li>
" +
      \`<li id="fn2" class="footnote-item"><p>Here's one with multiple blocks.</p>
\` +
      '<p>Subsequent paragraphs are indented to show that they <a href="#fnref2" class="footnote-backref">↩︎</a></p>
' +
      "</li>
" +
      "</ol>
" +
      "</section>
",
    data: {
      children: '<p>Here is a footnote reference,<sup class="footnote-ref"><a href="#fn1" id="fnref1">[1]</a></sup> and another.<sup class="footnote-ref"><a href="#fn2" id="fnref2">[2]</a></sup></p>
' +
        "<p>belong to the previous footnote.</p>
" +
        '<h4 class="mt-3">Footnotes</h4>
' +
        '<section class="footnotes">
' +
        '<ol class="footnotes-list">
' +
        '<li id="fn1" class="footnote-item"><p>Here is the footnote. <a href="#fnref1" class="footnote-backref">↩︎</a></p>
' +
        "</li>
" +
        \`<li id="fn2" class="footnote-item"><p>Here's one with multiple blocks.</p>
\` +
        '<p>Subsequent paragraphs are indented to show that they <a href="#fnref2" class="footnote-backref">↩︎</a></p>
' +
        "</li>
" +
        "</ol>
" +
        "</section>
",
      content: "Here is a footnote reference,[^1] and another.[^longnote]
" +
        "
" +
        "[^1]: Here is the footnote.
" +
        "
" +
        "[^longnote]: Here's one with multiple blocks.
" +
        "
" +
        "    Subsequent paragraphs are indented to show that they
" +
        "
" +
        "belong to the previous footnote.
",
      date: [],
      mergedKeys: [
        "tags",
      ],
      page: [
        "src",
        "data",
      ],
      paginate: "paginate",
      search: [],
      tags: "Array(0)",
      url: "/footnote/",
    },
    src: {
      asset: undefined,
      ext: ".md",
      path: "/footnote",
      remote: undefined,
      slug: "footnote",
    },
  },
  {
    content: '<!DOCTYPE html>
<p><a href="#foo" target="_blank">link</a></p>
',
    data: {
      children: '<p><a href="#foo" target="_blank">link</a></p>
',
      content: "[link](#foo){target=_blank}
",
      date: [],
      mergedKeys: [
        "tags",
      ],
      page: [
        "src",
        "data",
      ],
      paginate: "paginate",
      search: [],
      tags: "Array(0)",
      url: "/with-attributes/",
    },
    src: {
      asset: undefined,
      ext: ".md",
      path: "/with-attributes",
      remote: undefined,
      slug: "with-attributes",
    },
  },
  {
    content: "<!DOCTYPE html>
" +
      '<pre><code class="language-html">&lt;p&gt;Example&lt;/p&gt;
' +
      "</code></pre>
" +
      "<pre><code>&lt;p&gt;Example without color highlight&lt;/p&gt;
" +
      "</code></pre>
" +
      "<pre><code>Indented code
" +
      "</code></pre>
",
    data: {
      children: '<pre><code class="language-html">&lt;p&gt;Example&lt;/p&gt;
' +
        "</code></pre>
" +
        "<pre><code>&lt;p&gt;Example without color highlight&lt;/p&gt;
" +
        "</code></pre>
" +
        "<pre><code>Indented code
" +
        "</code></pre>
",
      content: "\`\`\`html
" +
        "<p>Example</p>
" +
        "\`\`\`
" +
        "
" +
        "\`\`\`
" +
        "<p>Example without color highlight</p>
" +
        "\`\`\`
" +
        "
" +
        "    Indented code
",
      date: [],
      mergedKeys: [
        "tags",
      ],
      page: [
        "src",
        "data",
      ],
      paginate: "paginate",
      search: [],
      tags: "Array(0)",
      url: "/with-code/",
    },
    src: {
      asset: undefined,
      ext: ".md",
      path: "/with-code",
      remote: undefined,
      slug: "with-code",
    },
  },
  {
    content: "<!DOCTYPE html>
" +
      "<!-- deno-fmt-ignore-file -->
" +
      "<dl>
" +
      "<dt>This is a definition title</dt>
" +
      "<dd>And this the description</dd>
" +
      "<dd>Other description</dd>
" +
      "<dt>Other title</dt>
" +
      "<dd>And other description</dd>
" +
      "</dl>
",
    data: {
      children: "<!-- deno-fmt-ignore-file -->
" +
        "<dl>
" +
        "<dt>This is a definition title</dt>
" +
        "<dd>And this the description</dd>
" +
        "<dd>Other description</dd>
" +
        "<dt>Other title</dt>
" +
        "<dd>And other description</dd>
" +
        "</dl>
",
      content: "<!-- deno-fmt-ignore-file -->
" +
        "This is a definition title
" +
        ": And this the description
" +
        ": Other description
" +
        "
" +
        "Other title
" +
        ": And other description
",
      date: [],
      mergedKeys: [
        "tags",
      ],
      page: [
        "src",
        "data",
      ],
      paginate: "paginate",
      search: [],
      tags: "Array(0)",
      url: "/with-deflist/",
    },
    src: {
      asset: undefined,
      ext: ".md",
      path: "/with-deflist",
      remote: undefined,
      slug: "with-deflist",
    },
  },
  {
    content: "<!DOCTYPE html>
" +
      "<h1>Module <strong>example</strong></h1>
" +
      "
" +
      '<div><p>Welcome to this <a href="/">page</a></p></div>
',
    data: {
      children: "<h1>Module <strong>example</strong></h1>
" +
        "
" +
        '<div><p>Welcome to this <a href="/">page</a></p></div>
',
      content: "<h1>{{ title | md(true) | safe }}</h1>
" +
        "
" +
        "<div>{{ description | md | safe }}</div>
",
      date: [],
      description: "Welcome to this [page](/)
",
      mergedKeys: [
        "tags",
      ],
      page: [
        "src",
        "data",
      ],
      paginate: "paginate",
      search: [],
      tags: "Array(0)",
      title: "Module **example**",
      url: "/with-filter/",
    },
    src: {
      asset: undefined,
      ext: ".njk",
      path: "/with-filter",
      remote: undefined,
      slug: "with-filter",
    },
  },
  {
    content: "<!DOCTYPE html>
" +
      "<h1>Module example</h1>
" +
      '<p><a href="/">Back to home</a></p>
',
    data: {
      children: '<h1>Module example</h1>
<p><a href="/">Back to home</a></p>
',
      content: "default",
      date: [],
      mergedKeys: [
        "tags",
      ],
      page: [
        "src",
        "data",
      ],
      paginate: "paginate",
      search: [],
      tags: "Array(0)",
      templateEngine: "tmpl.js,md",
      title: "Module example",
      url: "/with-module/",
    },
    src: {
      asset: undefined,
      ext: ".tmpl.js",
      path: "/with-module",
      remote: undefined,
      slug: "with-module",
    },
  },
  {
    content: "<!DOCTYPE html>
" +
      "<!-- deno-fmt-ignore-file -->
" +
      "<h1></h1>
" +
      "<p>Foo</p>
" +
      "<ul>
" +
      '<li>one: <a href="/items/un.html">un</a></li>
' +
      '<li>two: <a href="/items/dous.html">dous</a></li>
' +
      '<li>three: <a href="/items/tres.html">tres</a></li>
' +
      "</ul>
",
    data: {
      children: "<!-- deno-fmt-ignore-file -->
" +
        "<h1></h1>
" +
        "<p>Foo</p>
" +
        "<ul>
" +
        '<li>one: <a href="/items/un.html">un</a></li>
' +
        '<li>two: <a href="/items/dous.html">dous</a></li>
' +
        '<li>three: <a href="/items/tres.html">tres</a></li>
' +
        "</ul>
",
      content: "<!-- deno-fmt-ignore-file -->
" +
        "# {{ title }}
" +
        "
" +
        "Foo
" +
        "
" +
        "{% for title, no in myData %}
" +
        "- {{ title }}: [{{ no }}](/items/{{ no }}.html)
" +
        "{%- endfor %}
",
      date: [],
      mergedKeys: [
        "tags",
      ],
      myData: [
        "one",
        "two",
        "three",
      ],
      page: [
        "src",
        "data",
      ],
      paginate: "paginate",
      search: [],
      tags: "Array(0)",
      templateEngine: "njk,md",
      url: "/with-nunjucks/",
    },
    src: {
      asset: undefined,
      ext: ".md",
      path: "/with-nunjucks",
      remote: undefined,
      slug: "with-nunjucks",
    },
  },
]
`;
